app/javascript/utils/fetch-json.ts

```diff
@@ -5,23 +5,28 @@ export async function fetchJSON(endpoint: string, options: any) {
     'content-type': 'application/json',
     accept: 'application/json',
   }
-  const response = await fetch(
-    endpoint,
-    Object.assign(options, { headers: headers })
-  )
 
-  if (!response.ok) {
-    throw response
-  }
+  try {
+    const response = await fetch(
+      endpoint,
+      Object.assign(options, { headers: headers })
+    )
 
-  const contentType = response.headers.get('Content-Type')
-  if (
-    !contentType ||
-    (!contentType.includes('+json') &&
-      !contentType.includes('application/json'))
-  ) {
-    throw response
-  }
+    if (!response.ok) {
+      throw response
+    }
 
-  return camelizeKeys(await response.json())
+    const contentType = response.headers.get('Content-Type')
+    if (
+      !contentType ||
+      (!contentType.includes('+json') &&
+        !contentType.includes('application/json'))
+    ) {
+      throw response
+    }
+
+    return camelizeKeys(await response.json())
+  } catch (responseOrError) {
+    return Promise.reject(responseOrError)
```

Comment:
Errors thrown in `async` functions are automatically turned into promise. Remove the entire `try` `catch`:

```
async function hello(boom = false) {
  if (boom) {
    throw new Error('boom')
  }

  return 42
}

hello().then(console.log)
// => 42

hello().catch(console.error)
// => Error('boom')
```